riscv/src/lib.rs

```diff
@@ -198,10 +173,25 @@ macro_rules! as_ref [
     };
 ];
 
-pub fn compile_rust_crate_to_riscv(
-    input_dir: &str,
-    output_dir: &Path,
-) -> (Option<PathBuf>, BTreeMap<String, PathBuf>) {
+pub struct CompilationResult {
+    pub executable: Option<PathBuf>,
+    assemblies: BTreeMap<String, PathBuf>,
```

Comment:
why just make one `pub` and the other not?

Reply:
because there is no methog to get the `pub` one